view/SSViewer.php

```diff
@@ -536,7 +536,7 @@ public function getInjectedValue($property, $params, $cast = true) {
 					// If not provided, use default
 					if (!$casting) $casting = Config::inst()->get('ViewableData', 'default_cast', Config::FIRST_SET);
 
-					$obj = new $casting($property);
+					$obj = Injector::inst()->get($casting, false, array($property));
```

Comment:
Shall we just use ->create() here? We normally use ->get() for singletons (just for consistency of style).